### ISSUE TITLE
left join generate wrong non null field

### ISSUE THREAD

**Issue by YeungKC** [simolus3/moor#1323]
I have a custom result query, and when I upgraded to moor 4.4.0, some fields that were nullable became non null, but in fact he was null.

sql:
```sql

messageItems AS MessageItem:
SELECT m.message_id AS messageId, m.conversation_id AS conversationId, u.user_id AS userId, c.owner_id AS conversationOwnerId, c.category AS conversionCategory,
        u.full_name AS userFullName, u.identity_number AS userIdentityNumber, u.app_id AS appId, m.category AS type,
        m.content AS content, m.created_at AS createdAt, m.status AS status, m.media_status AS mediaStatus, m.media_waveform AS mediaWaveform,
        m.name AS mediaName, m.media_mime_type AS mediaMimeType, m.media_size AS mediaSize, m.media_width AS mediaWidth, m.media_height AS mediaHeight,
        m.thumb_image AS thumbImage, m.thumb_url AS thumbUrl, m.media_url AS mediaUrl, m.media_duration AS mediaDuration, m.quote_message_id as quoteId,
        m.quote_content as quoteContent, u1.full_name AS participantFullName, m.action AS actionName, u1.user_id AS participantUserId,
        s.snapshot_id AS snapshotId, s.type AS snapshotType, s.amount AS snapshotAmount, a.symbol AS assetSymbol, s.asset_id AS assetId,
        a.icon_url AS assetIcon, st.asset_url AS assetUrl, st.asset_width AS assetWidth, st.asset_height AS assetHeight, st.sticker_id AS stickerId,
        st.name AS assetName, st.asset_type AS assetType, h.site_name AS siteName, h.site_title AS siteTitle, h.site_description AS siteDescription,
        h.site_image AS siteImage, m.shared_user_id AS sharedUserId, su.full_name AS sharedUserFullName, su.identity_number AS sharedUserIdentityNumber,
        su.avatar_url AS sharedUserAvatarUrl, su.is_verified AS sharedUserIsVerified, su.app_id AS sharedUserAppId, mm.has_read as mentionRead,
        c.name AS groupName, u.relationship AS relationship, u.avatar_url AS avatarUrl
        FROM messages m
        INNER JOIN users u ON m.user_id = u.user_id
        LEFT JOIN users u1 ON m.participant_id = u1.user_id
        LEFT JOIN snapshots s ON m.snapshot_id = s.snapshot_id
        LEFT JOIN assets a ON s.asset_id = a.asset_id
        LEFT JOIN stickers st ON st.sticker_id = m.sticker_id
        LEFT JOIN hyperlinks h ON m.hyperlink = h.hyperlink
        LEFT JOIN users su ON m.shared_user_id = su.user_id
        LEFT JOIN conversations c ON m.conversation_id = c.conversation_id
        LEFT JOIN message_mentions mm ON m.message_id = mm.message_id
        WHERE $where
        ORDER BY $order
        LIMIT $limit;

```

fields:
```
  final String snapshotId;
  final String snapshotType;
  final String snapshotAmount;
  final String assetSymbol;
  final String assetId;
  final String assetIcon;
  final String assetUrl;
  final int assetWidth;
  final int assetHeight;
  final String stickerId;
  final String assetName;
  final String assetType;
  final String siteName;
  final String siteTitle;
```

**Comment by simolus3** [comment]
Well, this is embarrassing 😬 I tried to improve nullability analysis in the last release and all tests somehow failed to detect this regression.

This should be fixed in c12de0b8a9035cb61c3192ad7ae020df5fe62e5e, can you try that? You can add this to your pubspec, then run `flutter pub run build_runner clean` followed by another build:

```yaml
dependency_overrides:
  sqlparser:
    git:
      url: https://github.com/simolus3/moor.git
      ref: develop
      path: sqlparser
```

I hope that fixes the problem, if it does I'll release a fix ASAP.

**Comment by YeungKC** [comment]
It works as expected after using sqlparser on the develop branch.

**Comment by simolus3** [comment]
Great! I've just published sqlparser version 0.17.1 contains this fix.
